File: service/src/main/java/hu/bsstudio/race/speed/configuration/SpeedRaceServiceConfiguration.java
package hu.bsstudio.race.speed.configuration;

import hu.bsstudio.race.speed.DefaultSpeedRaceService;
import hu.bsstudio.race.speed.SpeedRaceService;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SpeedRaceServiceConfiguration {

    @Bean
    public SpeedRaceService speedRaceService() {
        return new DefaultSpeedRaceService();
    }
}

File: server/service/src/main/kotlin/hu/bsstudio/robonaut/race/speed/configuration/SpeedRaceServiceConfiguration.kt
package hu.bsstudio.robonaut.race.speed.configuration

import hu.bsstudio.robonaut.race.speed.BroadcastingSpeedRaceService
import hu.bsstudio.robonaut.race.speed.DefaultSpeedRaceService
import hu.bsstudio.robonaut.race.speed.SpeedRaceService
import hu.bsstudio.robonaut.repository.TeamRepository
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SpeedRaceServiceConfiguration(
    private val repository: TeamRepository,
    private val rabbitTemplate: RabbitTemplate,
) {
    @Bean
    fun speedRaceService(defaultSpeedRaceService: SpeedRaceService): SpeedRaceService =
        BroadcastingSpeedRaceService(rabbitTemplate, defaultSpeedRaceService)

    @Bean
    fun defaultSpeedRaceService(): SpeedRaceService = DefaultSpeedRaceService(repository)
}
